include/zenoh-pico/api/primitives.h

```diff
@@ -1133,6 +1133,49 @@ const char *z_string_data(const z_loaned_string_t *str);
  */
 size_t z_string_len(const z_loaned_string_t *str);
 
+/**
+ * Builds a :c:type:`z_string_t` by wrapping a ``const char *`` string.
+ *
+ * Parameters:
+ *   str: Pointer to an uninitialized :c:type:`z_string_t`.
+ *   value: Pointer to a null terminated string.
+ *
+ * Return:
+ *   ``0`` if creation successful, ``negative value`` otherwise.
+ */
+int8_t z_string_wrap(z_owned_string_t *str, const char *value);
```

Comment:
there are some concerns about naming - see https://github.com/eclipse-zenoh/zenoh-c/issues/475
Maybe it's good moment to find some solution and then implement it on zenoh-c side